netdiff/parsers/openvpn.py

```diff
@@ -0,0 +1,67 @@
+import networkx
+from openvpn_status import ParsingError, parse_status
+
+from ..exceptions import ConversionException, ParserError
+from .base import BaseParser
+
+
+class OpenvpnParser(BaseParser):
+    """ OpenVPN status log parser """
+    protocol = 'openvpn'
+    version = ''
+
+    def to_python(self, data):
+        """
+        Adds support for txtinfo format
+        """
+        try:
+            return super(OpenvpnParser, self).to_python(data)
+        except ConversionException as e:
+            return self._txtinfo_to_json(e.data)
+
+    def parse(self, data):
+        """
+        Converts a OpenVPN JSON to a NetworkX Graph object
+        which is then returned.
+        """
+        # initialize graph and list of aggregated nodes
+        graph = networkx.graph()
+        if 'type' not in data or data['type'] != 'OpenVPN':
+            raise ParserError('Parse Error, not a OpenVPN object')
+        required_keys = ['nodes', 'serverIp']
+        for key in required_keys:
+            if key not in data:
+                raise ParserError('Parse Error, "%s" key not found' % key)
+
+        graph.add_node(data['serverIp'])
+        for link in nodes:
+            try:
+                source = data['serverIp']
+                target = link['realAddress']
+            except KeyError as e:
+                raise ParserError('Parse Error, "%s" key not found' % key)
+            graph.add_node(target)
+            graph.add_edge(source, target, weight=1)
+        return graph
+
+    def _txtinfo_to_jsoninfo(self, data):
```

Comment:
this is the method that should be called `to_python` (you can get rid of the other one).
You choose a starting point of another parser which was a special case.

Since you already have a library that parses the data, you don't really need to do much apart returning the data structure that the parser expects

Reply:
I think you are right. We don't really need this method. Will edit